src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,47 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_STORAGE_LENGTH = 10;
+    private final Pair<K, V>[] storage;
+    private int size;
+
+    public StorageImpl() {
+        storage = new Pair[MAX_STORAGE_LENGTH];
+    }
+
     @Override
     public void put(K key, V value) {
+        if (size == MAX_STORAGE_LENGTH) {
+            throw new ArrayIndexOutOfBoundsException("Array already full");
+        }
+        if (getIndexByKey(key) >= 0) {
```

Comment:
```suggestion
        int index = getIndexByKey(key);
        if (index >= 0) {
             storage[index] = new Pair<>(key, value);
```

don't need to call method twice